tern/classes/docker_image.py

```diff
@@ -159,3 +167,20 @@ def load_image(self, load_until_layer=0):
                 e.returncode, cmd=e.cmd, output=e.output, stderr=e.stderr)
         except IOError as e:
             raise IOError(e)
+        if self.load_until_layer > self.total_layers:
+            raise LayerNumberError(self.load_until_layer, self.total_layers)
```

Comment:
When I tested this exception I see:

```
2021-02-10 22:07:56,723 - DEBUG - rootfs - Running command: sudo /home/rjudge/tt/tern/tern/tools/fs_hash.sh /home/rjudge/.tern/temp/44dbe4be81909effaf6bfe136f9251becf14363449d14a7d0a162ccc3ae7ef84/contents
2021-02-10 22:07:56,760 - WARNING - image - Given layer: 23 exceeds total amount of layers: 8 in image.
2021-02-10 22:07:56,761 - DEBUG - common - Loading packages from cache: layer "96d121b7df"
2021-02-10 22:07:57,030 - DEBUG - common - Loading packages from cache: layer "f3f7474351"
2021-02-10 22:08:03,080 - DEBUG - common - Reading files in filesystem...
```
Which is fine, but the final report only looks like this (with no further layer information):
```
This report was generated by the Tern Project
https://github.com/tern-tools/tern/commit/3d6333088dfa900316eeef606ca7028dae48f6a8

Docker image: golang:latest:

```

If the user gives a layer that is higher than the total number of layers, we should print the entire report.
Also, I think I missed this or misunderstood your question about this in our slack discussion so I apologize but creating a `LayerNumberError` exception is probably unnecessary here...

For this particular use case, instead of alerting the user in docker_image.py that they put an incorrect number of layers, could we just silently load all the layers and then print something in the default `generate()` method using the `logger` in `tern/formats/default/generator.py`? The reason I say that is:
1) I think the warning message using the exception here gets lost in the console output. If we move it to the generate method, there is logger text that says "Creating a detailed report of components in image..." and here we could also add something like you suggested - `"Given layer <load_until_layer> exceeds total number of layers in image (<total_layers>). Ignoring --layer option and generating report for <total_layers> total layers"` 
2) The user doesn't really care or know anything about the concept of "loading" layers. They are using the `--layer` option thinking that it only changes the output report we get but don't need to know what that looks like from an implementation POV. To group the console/log warning message in this case with the reporting seems to be more intuitive for the user.

Reply:
thought the way I made it would be easiest to maintain (as all layer limiting is done in DockerImage.load_image, thats where I'd expect throwing err about it)
Instead of adding it to generate() - which would require adding same log for each format,  (and would be easy to miss when adding new format) i went with `tern/report/report.py`'s `report_out` as it takes all args as argument. Added a comment also so its not confusing in the future.